src/backend/marsha/core/tests/test_models_video.py

```diff
@@ -38,3 +42,31 @@ def test_models_video_fields_lti_id_unique(self):
         # Soft deleted videos should not count for unicity
         video.delete(force_policy=SOFT_DELETE_CASCADE)
         VideoFactory(lti_id=video.lti_id, playlist=video.playlist)
+
+    def test_models_video_is_ready_to_show(self):
+        """All combination where a video is ready or not to be shown."""
+        # Test all state choices allowing to be ready to show
+        for state_choice in STATE_CHOICES:
+            video = VideoFactory(
+                upload_state=state_choice[0],
+                uploaded_on=datetime(2018, 8, 8, tzinfo=pytz.utc),
+            )
+
+            self.assertEqual(
+                video.is_ready_to_show,
+                state_choice[0] not in [DELETED, HARVESTED],
+            )
+
+        # when uploaded_on is null, the video can not be ready
+        for state_choice in STATE_CHOICES:
+            video = VideoFactory(
+                upload_state=state_choice[0],
+            )
```

Comment:
It may be more explicit if we add uploaded_on here:
```suggestion
            video = VideoFactory(
                upload_state=state_choice[0],
                uploaded_on=null,
            )
```

Reply:
`None` 😛

Reply:
Bitten by js :face_with_head_bandage: